heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/sink/WebSink.java

```diff
@@ -0,0 +1,162 @@
+ */
+public class WebSink implements IMetricsSink {
+  private static final Logger LOG = Logger.getLogger(WebSink.class.getName());
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private static final int HTTP_STATUS_OK = 200;
+
+  private static final String KEY_PORT = "port";
+  private static final String KEY_PORT_FILE = "port-file";
+  private static final String KEY_PATH = "path";
+  private static final String KEY_FLAT_METRICS = "flat-metrics";
+  private static final String KEY_INCLUDE_TOPOLOGY_NAME = "include-topology-name";
+
+  protected ConcurrentHashMap<String, Object> metrics = new ConcurrentHashMap<>();
+  private HttpServer httpServer;
+  private boolean isFlatMetrics = true;
+  private boolean includeServiceNamespace = true;
+  private String topologyName;
+
+  @Override
+  public void init(Map<String, Object> conf, SinkContext context) {
+    String path = (String) conf.get(KEY_PATH);
+    String portFile = (String) conf.get(KEY_PORT_FILE);
+    isFlatMetrics = TypeUtils.getBoolean(conf.getOrDefault(KEY_FLAT_METRICS, true));
+    includeServiceNamespace = TypeUtils.getBoolean(
+        conf.getOrDefault(KEY_INCLUDE_TOPOLOGY_NAME, false));
+    topologyName = context.getTopologyName();
+
+    int port = TypeUtils.getInteger(conf.getOrDefault(KEY_PORT, 0));
+    if (port == 0) {
+      if (portFile != null) {
+        try {
+          port = TypeUtils.getInteger(Files.lines(Paths.get(portFile)).findFirst().get().trim());
+        } catch (IOException e) {
+          throw new IllegalArgumentException("Could not parse " + KEY_PORT_FILE + " " + portFile
+              + " Make sure the file only contains the port on which the service should run"
+              + " and is UTF8 encoded", e);
+        }
+      } else {
+        throw new IllegalArgumentException("Neither 'port' nor 'port_file' "
+            + "were specified in config for metrics sink " + context.getSinkId());
+      }
+    }
+    startHttpServer(path, port);
+  }
+
+  /**
+   * Start a http server on supplied port that will serve the metrics, as json,
+   * on the specified path.
+   *
+   * @param path
+   * @param port
+   */
+  protected void startHttpServer(String path, int port) {
+    try {
+      httpServer = HttpServer.create(new InetSocketAddress(port), 0);
+      httpServer.createContext(path, new HttpHandler() {
+        @Override
+        public void handle(HttpExchange httpExchange) throws IOException {
+          byte[] response = MAPPER.writeValueAsString(metrics).getBytes();
+          httpExchange.sendResponseHeaders(HTTP_STATUS_OK, response.length);
+          OutputStream os = httpExchange.getResponseBody();
+          os.write(response);
+          os.close();
+          LOG.log(Level.INFO, "Received metrics request.");
+        }
+      });
+      httpServer.start();
+    } catch (IOException e) {
+      LOG.log(Level.SEVERE, "Could not create HttpServer on port " + port, e);
```

Comment:
Given the RuntimeException contains the error msg, extra SEVERE LOG seems duplicated and unnecessary.